RecoParticleFlow/PFClusterProducer/plugins/PFClusterFromHGCalMultiCluster.cc

```diff
@@ -14,43 +17,73 @@ void PFClusterFromHGCalMultiCluster::buildClusters(const edm::Handle<reco::PFRec
   const auto& hgcalMultiClusters = *clusterH_;
   auto const& hits = *input;
 
+  const auto& tracksters = *trackstersH_;
+
   // for quick indexing back to hit energy
   std::unordered_map<uint32_t, size_t> detIdToIndex(hits.size());
   for (uint32_t i = 0; i < hits.size(); ++i) {
     detIdToIndex[hits[i].detId()] = i;
   }
 
+  int iMultiClus = -1;
+
   for (const auto& mcl : hgcalMultiClusters) {
+    iMultiClus++;
+
+    // Filter using trackster PID
+    if (filterByTracksterPID_) {
+      double probTotal = 0;
+
+      for (int iCat = 0; iCat < (int)filter_on_categories_.size(); iCat++) {
+        int cat = filter_on_categories_[iCat];
+
+        double prob = tracksters[iMultiClus].id_probabilities(cat);
```

Comment:
`id_probabilities()` is a `float:` therefore, there is no need to make `prob` a `double` (and as a consequence `probTotal` and the class member `pid_threshold_`): better to have all them as `float`